packages/vite/src/module-runner/types.ts

```diff
@@ -155,7 +151,7 @@ export interface ModuleRunnerOptions {
   /**
    * A set of methods to communicate with the server.
    */
-  transport: RunnerTransport
+  createTransport: CreateRunnerTransport
```

Comment:
Personally, I hate passing down the function here. Maybe we can pass down `connect` handler alongside `close` handler?

Reply:
I don't know why I didn't come up with that. I didn't like it either 😅 Done in https://github.com/vitejs/vite/pull/18362/commits/313fac8339bdba879153eca11dd1c262973d551f